### PR TITLE
Fixes datetime pickers

### PR DESCRIPTION
**Pull request by cyrillefr**
**Checklist**

- [x] I have read the [Contribution & Best practices Guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md).
- [x] My branch is up-to-date with the upstream `master` branch.
- [x] The tests pass locally with my changes.

**Short description of what this resolves/which [issues](https://github.com/openSUSE/osem/issues) does this fix?:**

- read only attributes in haml to prevent non valid manual date
- typos in haml and in JS
- undefined methods in JS

- Fixes https://github.com/openSUSE/osem/issues/2455

**Changes proposed in this pull request:**

- Add readonly on datetime pickers (haml)
- Fix some typos (haml)
- Update method names in JavaScrip that do not exit for bootstrap component
- Do some name matching between in JS and Haml

### CODE REVIEW

File: app/assets/javascripts/osem-datepickers.js [hunk 1]
```diff
@@ -4,8 +4,8 @@ $(function () {
   $("input[id^='datetimepicker']").datetimepicker({
     useCurrent: false,
     sideBySide: true,
-    autoclose: true,
-    format: 'YYYY-MM-DD HH:mm'
+    format: 'YYYY-MM-DD HH:mm',
+    ignoreReadonly: true
```
Comment (Ana06) [inline app/assets/javascripts/osem-datepickers.js]: why is this needed? where is readonly set? why not removing it there?

File: app/views/admin/conferences/new.html.haml [hunk 2]
```diff
@@ -9,7 +9,7 @@
                                 input_html: { required: 'required', pattern: '[a-zA-Z0-9_-]+', title: 'Only letters, numbers, underscores, and dashes.' }, prepend: conferences_url + '/'
       = f.inputs 'Scheduling' do
         = f.input :timezone, as: :time_zone, default: Time.zone.name, hint: 'Please select in what time zone your conference will take place.'
-        = f.input :start_date, as: :string, input_html: { id: 'conference-start-datepicker', required: 'required' }
-        = f.input :end_date, as: :string, input_html: { id: 'conference-end-datepicker', required: 'required' }
+        = f.input :start_date, as: :string, input_html: { id: 'conference-start-datepicker', required: 'required', readonly: 'readonly' }
```
Comment (Ana06) [inline app/views/admin/conferences/new.html.haml]: why do you set readonly?

File: app/assets/javascripts/osem-datepickers.js (hunk 1 above)
Comment (cyrillefr) [reply app/assets/javascripts/osem-datepickers.js]: Hi, this is is a 2 ways strategy.
By setting readonly in the haml, I prevent the user from entering non valid dates. But, by doing so, updating from the datetimepicker will not work. So, I added ignoreReadonly, so that user still cannot have access manually to the input, but the datetime picker has access to it.
cf. http://eonasdan.github.io/bootstrap-datetimepicker/Options/#ignorereadonly

File: app/views/admin/conferences/new.html.haml (hunk 2 above)
Comment (cyrillefr) [reply app/views/admin/conferences/new.html.haml]: To prevent user to input manually a date. And to be coherent with the first form I encountered (Registration Period) that was already read only.
To insure however that the datetime picker can update with a readonly attribute in html, I had to add an ignoreReadonly in the datepicker js file. (See my comment above).

Comment (Ana06) [reply app/views/admin/conferences/new.html.haml]: I wouldn't prevent the user to manually input a date. I would make it coherent the other way around.

Comment (cyrillefr) [reply app/views/admin/conferences/new.html.haml]: Ok, I will update the changes in that way and remove the readonly attributes.

Comment (cyrillefr) [reply app/views/admin/conferences/new.html.haml]: I have pushed the changes. I hope this is ok.

**Comment by differentreality** [comment]
Are pickers working for you now?
